### PR TITLE
Implemented the Bus Route, Draw polyline feat, and hidden unrelated stops

### LINKED ISSUE #30: Hide unrelated stops on the map while a route line is being displayed
## User story

As a user that has selected a particular route,
I don't want to see stops that are not part of the route,
so that I can focus on my current route

## Acceptance criteria

- [x] Hide stops that are not part of the currently selected route
- [x] Show all stops from the currently selected route
- [x] Show all stops after route selection is cleared

### PR DESCRIPTION
**Pull request by tarunsinghofficial**
Fixes #30 #29 #24 issues.

### CODE REVIEW

**Comment by tarunsinghofficial** [comment]
The feature is working correctly but there's one issue left. When we try to choose a new stop, the old bus route is not removed. So, working on the leftover issue now.

File: src/components/map/GoogleMap.svelte [hunk 1]
```diff
@@ -51,20 +56,54 @@
 	async function loadStopsAndAddMarkers(lat, lng) {
 		const json = await loadStopsForLocation(lat, lng);
 		const stops = json.data.list;
+		allStops = [...allStops, ...stops];
```
Comment (aaronbrethorst) [inline src/components/map/GoogleMap.svelte:59]: be sure to de-dupe these lists as you merge them.

File: src/components/map/GoogleMap.svelte [hunk 2]
```diff
@@ -51,20 +56,54 @@
...
+			stops.forEach((s) => addMarker(s));
 		}
 	}
 
-	function markerExists(s) {
-		return markers.some((marker) => marker.s.id === s.id);
+	function clearAllMarkers() {
+		markers.forEach(({ marker, overlay, element }) => {
+			if (marker) {
+				marker.setMap(null);
```
Comment (aaronbrethorst) [inline src/components/map/GoogleMap.svelte:74]: this can be simplified to `marker?.setMap(null)`

File: src/components/map/GoogleMap.svelte [hunk 3]
```diff
@@ -51,20 +56,54 @@
...
+			}
+			if (overlay) {
```
Comment (aaronbrethorst) [inline src/components/map/GoogleMap.svelte:76]: same here

File: src/components/map/GoogleMap.svelte [hunk 4]
```diff
@@ -51,20 +56,54 @@
...
+					overlay.map = null;
+				}
+				if (overlay.draw) {
+					overlay.draw = () => {};
+				}
+				if (overlay.onRemove) {
+					overlay.onRemove();
+				}
+			}
+			if (element && element.parentNode) {
```
Comment (aaronbrethorst) [inline src/components/map/GoogleMap.svelte:87]: same here

File: src/components/map/GoogleMap.svelte [hunk 5]
```diff
@@ -51,20 +56,54 @@
...
+	$: if (selectedRoute && showRoute) {
+		clearAllMarkers();
+		const stopsToShow = allStops.filter((s) => s.routeIds.includes(selectedRoute.id));
+		stopsToShow.forEach((s) => addMarker(s));
+	} else if (!showRoute || !selectedRoute) {
+		clearAllMarkers();
+		allStops.forEach((s) => addMarker(s));
+	}
+
+	/* function markerExists(s) {
```
Comment (aaronbrethorst) [inline src/components/map/GoogleMap.svelte:103]: delete me

File: src/components/map/GoogleMap.svelte [hunk 6]
```diff
@@ -107,6 +146,11 @@
 			container.style.position = 'absolute';
 			this.getPanes().overlayMouseTarget.appendChild(container);
 		};
+		overlay.onRemove = function () {
+			if (container.parentNode) {
```
Comment (aaronbrethorst) [inline src/components/map/GoogleMap.svelte:150]: use optional chaining here too to get rid of the `if` statement

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
I've noted a few places where code can be simplified or deleted but this is looking good!

File: src/components/map/GoogleMap.svelte (hunk 2 above)
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:74]: sure, done the optional chaining.

File: src/components/map/GoogleMap.svelte (hunk 3 above)
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:76]: done

File: src/components/map/GoogleMap.svelte (hunk 4 above)
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:87]: done

File: src/components/map/GoogleMap.svelte (hunk 5 above)
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:103]: deleted the component

File: src/components/map/GoogleMap.svelte (hunk 6 above)
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:150]: removed the `if` statement in place of optional chaining.

File: src/components/map/GoogleMap.svelte (hunk 1 above)
Comment (tarunsinghofficial) [reply src/components/map/GoogleMap.svelte:59]: sure, done.

**Pull request merged**